Client/Frontend/Onboarding/Models/IntroViewModel.swift

```diff
@@ -44,4 +58,15 @@ class IntroViewModel: OnboardingViewModelProtocol, FeatureFlaggable {
     func saveHasSeenOnboarding() {
         introScreenManager?.didSeeIntroScreen()
     }
+
+    // MARK: SkAdNetwork
+    // this event can only be sent once in this time window
```

Comment:
There was a miscommunication here, and I think it was on me. When we saw "the conversion value can only be sent once", that's saying that it gets locked in and sent by _Apple_, not be us. We are free to update the conversion value as many times as we like. If there is a 24-hour period where it is _not_ updated, that's when Apple locks it in and sends it.

Because of that, we can do as we did previously in this PR and update the conversion value immediately after each click (down in the `IntroViewController`). That way, the conversion value is guaranteed to be updated without any delay (if e.g. someone clicked on the first one then swiped out).

Reply:
@fbertsch No worries, thank you for taking the time to check this while on vacation, should I revert to the previous [commit](https://github.com/mozilla-mobile/firefox-ios/pull/16099/commits/db9ff605c2d5acbbc60ec8ef648b96f1c2a2685d) that had this implementation?

Reply:
@fbertsch should it be a combo of the two where the final value can be between 1-7, where 1,2,3 would mean the user only selected one action while a larger sum would mean a user has selected a combination of onboarding actions? Or the first implementation just that instead of a plain integer it sends the bitset?

Reply:
The _current_ implementation is correct in what we want the values to be, where each bit is associated with a different action. However we want the value to be updated each time the user takes an action (immediately after), as you had in the previous commit.

So if a user selects the first onboarding action, values is `1`. If they then select the second action as well, value is updated to `3`. If they did _not_ interact with the first action, then they interacted with the second, the value would be set to `2`, and so on.

Does that help clarify? We basically don't want a situation where users are performing an action, but the conversion value doesn't get set.

Reply:
@fbertsch yes, thank you, that clarifies the cases I was thinking about